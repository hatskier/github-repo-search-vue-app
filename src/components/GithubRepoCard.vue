<template>
  <v-card
    class="mx-auto"
    max-width="600"
    outlined
  >
    <v-list-item three-line>
      
      <v-list-item-content>
        <v-list-item-title class="headline mb-1">
          Value 1
        </v-list-item-title>
        <div class="mb-8">
          Value 2
        </div>

        <v-list-item-subtitle>
          Value 3
        </v-list-item-subtitle>

        <div class="mt-2 mb-2">
          <v-list-item-subtitle>
            Value 4
          </v-list-item-subtitle>
        </div>
      </v-list-item-content>
    </v-list-item>

    <v-card-actions>
      <v-btn
        color="blue"
        text
      >
        ACTION
        <v-icon
          small
          class="pa-2"
        >
          mdi-pencil
        </v-icon>
      </v-btn>
    </v-card-actions>
  </v-card>
</template>

<script>
export default {

}
</script>

<style>

</style>