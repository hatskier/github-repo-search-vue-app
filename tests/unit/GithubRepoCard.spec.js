// Libraries
import Vuetify from 'vuetify'

// Components
import GithubRepoCard from '@/components/GithubRepoCard.vue'

// Utilities
import { mount, createLocalVue } from '@vue/test-utils'
import './setup'

const localVue = createLocalVue()

describe('GithubRepoCard.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const mountFunction = options => {
    return mount(GithubRepoCard, {
      localVue,
      vuetify,
      ...options,
    })
  }

  it('should render main props', () => {
    const name = 'Test repo name'
    const stars = 10
    const owner = 'Hehehe'
    const openIssues = 20

    const wrapper = mountFunction({
      propsData: {
        name,
        stars,
        owner,
        openIssues,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    const title = wrapper.find('.v-list-item__title')
    expect(title.text()).toBe(name)
    const starsSpan = wrapper.find('span#stars')
    expect(starsSpan.text()).toBe(stars.toString())
    const openIssuesSpan = wrapper.find('span#open-issues')
    expect(openIssuesSpan.text()).toBe(openIssues.toString())
  })

  it('should render owner and repo links', () => {
    const name = 'Nice repo'
    const url = 'https://example.com/nice-repo'
    const owner = 'Haha'
    const ownerUrl = 'https://example.com/nice-owner-url'

    const wrapper = mountFunction({
      propsData: {
        name,
        url,
        owner,
        ownerUrl
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

    // Owner link
    const ownerLink = wrapper.find('a#owner-link')
    expect(ownerLink.text()).toBe(owner)
    expect(ownerLink.attributes('href')).toBe(ownerUrl)

    // Repo link
    const repoLink = wrapper.find('a#repo-link')
    expect(repoLink.attributes('href')).toBe(url)
  })
})
