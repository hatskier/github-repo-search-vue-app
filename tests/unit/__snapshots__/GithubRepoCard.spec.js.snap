// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubRepoCard.vue renders main props when passed 1`] = `
<div class="mx-auto v-card v-sheet v-sheet--outlined theme--light" style="max-width: 600px;">
  <div tabindex="-1" class="v-list-item v-list-item--three-line theme--light">
    <div class="v-list-item__content">
      <div class="v-list-item__title headline mb-1">
        Test repo name
      </div>
      <div class="mb-8">
        by
        <a id="owner-link" target="_blank">
          Hehehe
        </a></div>
      <div class="v-list-item__subtitle"><span class="v-tooltip v-tooltip--bottom"><!----><i aria-hidden="true" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate material-icons theme--light">star</i></span> <span id="stars">
          10
        </span></div>
      <div class="mt-2 mb-2">
        <div class="v-list-item__subtitle"><span class="v-tooltip v-tooltip--bottom"><!----><i aria-hidden="true" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate material-icons theme--light">info</i></span> <span id="open-issues">
            20
          </span></div>
      </div>
    </div>
    <div class="v-avatar v-list-item__avatar rounded rounded-0 v-avatar--tile grey" style="height: 80px; min-width: 80px; width: 80px;">
      <!----> <i aria-hidden="true" class="v-icon notranslate mdi mdi-account-circle theme--dark" style="font-size: 70px;"></i></div>
  </div>
  <div class="v-card__actions"><button target="_blank" type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default blue--text" id="repo-link"><span class="v-btn__content">
      OPEN
      <i aria-hidden="true" class="v-icon notranslate pa-2 material-icons theme--light" style="font-size: 16px;">open_in_new</i></span></button></div>
</div>
`;

exports[`GithubRepoCard.vue should render owner and repo links 1`] = `
<div class="mx-auto v-card v-sheet v-sheet--outlined theme--light" style="max-width: 600px;">
  <div tabindex="-1" class="v-list-item v-list-item--three-line theme--light">
    <div class="v-list-item__content">
      <div class="v-list-item__title headline mb-1">
        Nice repo
      </div>
      <div class="mb-8">
        by
        <a href="https://example.com/nice-owner-url" id="owner-link" target="_blank">
          Haha
        </a></div>
      <div class="v-list-item__subtitle"><span class="v-tooltip v-tooltip--bottom"><!----><i aria-hidden="true" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate material-icons theme--light">star</i></span> <span id="stars">

        </span></div>
      <div class="mt-2 mb-2">
        <div class="v-list-item__subtitle"><span class="v-tooltip v-tooltip--bottom"><!----><i aria-hidden="true" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate material-icons theme--light">info</i></span> <span id="open-issues">

          </span></div>
      </div>
    </div>
    <div class="v-avatar v-list-item__avatar rounded rounded-0 v-avatar--tile grey" style="height: 80px; min-width: 80px; width: 80px;">
      <!----> <i aria-hidden="true" class="v-icon notranslate mdi mdi-account-circle theme--dark" style="font-size: 70px;"></i></div>
  </div>
  <div class="v-card__actions"><a href="https://example.com/nice-repo" target="_blank" class="v-btn v-btn--flat v-btn--text theme--light v-size--default blue--text" id="repo-link"><span class="v-btn__content">
      OPEN
      <i aria-hidden="true" class="v-icon notranslate pa-2 material-icons theme--light" style="font-size: 16px;">open_in_new</i></span></a></div>
</div>
`;
